src/connector/src/sink/mod.rs

```diff
@@ -130,7 +130,7 @@ pub struct SinkWriterParam {
 
 #[async_trait]
 pub trait Sink {
-    type Writer: SinkWriter<CommitMetadata = ()>;
+    type Writer: SinkWriter;
```

Comment:
Why the associated type bound is removed?

Reply:
If we wrap the `IcebergWriter` with the `CoordinatedSinkWriter`, we don't have to remove the associated type bound here.